build_macos.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+set -e
+
+if ! command -v wget 2>&1 >/dev/null; then
+    echo "ERROR: Missing prerequisite: wget"
+    exit 1
+fi
+
+if ! command -v unzip 2>&1 >/dev/null; then
+    echo "ERROR: Missing prerequisite: unzip"
+    exit 1
+fi
+
+# Linux: https://corretto.aws/downloads/resources/21.0.2.13.1/amazon-corretto-21.0.2.13.1-linux-x64-jdk.zip
+# OSX: https://corretto.aws/downloads/resources/21.0.5.11.1/amazon-corretto-21.0.5.11.1-macosx-x64.tar.gz
+
+# Make sure that we actually have game files first
+if [ ! -f ./ufo50/data.win ]; then
+	echo "Place your UFO 50 game files in the ufo50 folder first!"
+	read -n 1 -s -r -p "Press any key to continue..."
+	echo
+	exit 1
+fi
+
+# Permissions
+chmod +x ./bin/zipalign-osx
+chmod +x ./bin/aapt-osx
+
+# Clean up from any previous runs
+rm -f ./UFO50Wrapper.apk
+rm -f ./com.unofficial.ufo50.zip
+rm -rf ./assets
+mkdir ./assets
+
+# Copy required assets
+echo "Preparing assets..."
+cp -r ./ufo50/ext ./assets/ext
+cp -r ./ufo50/Textures ./assets/Textures
+cp ./ufo50/*.dat ./assets/
+cp ./ufo50/options.ini ./assets/
+cp ./ufo50/data.win ./assets/game.droid
+cp ./base/AndroidWrapper2024.8.0.216_VM_debug.apk ./UFO50Wrapper.apk
+
+# Download Java, if needed
+if [ ! -f ./bin/java/Contents/Home/bin/java ]; then
+	ARCH="x64"
+	if [[ $(uname -m) == 'arm64' ]]; then
```

Comment:
The android binaries are universal binaries and with this here it should work on x86 and arm just fine. (only tested on arm)